src/main.cpp

```diff
@@ -979,27 +979,47 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
         // than the ones it replaces.
         CAmount nConflictingFees = 0;
         size_t nConflictingSize = 0;
+        CTxMemPool::setEntries allConflicting;
         if (setConflicts.size())
         {
             LOCK(pool.cs);
 
             // Check if it's economically rational to mine this transaction
-            // rather than the ones it replaces. For efficiency we simply sum
-            // up the pre-calculated fees/size-with-descendants values from the
-            // mempool package tracking; this does mean the pathological case
-            // of diamond tx graphs will be overcounted.
-            BOOST_FOREACH(const uint256 hashConflicting, setConflicts)
+            // rather than the ones it replaces.
+            const int maxDescendantsToVisit = 100;
+            BOOST_FOREACH(const uint256 &hashConflicting, setConflicts)
             {
                 CTxMemPool::txiter mi = pool.mapTx.find(hashConflicting);
                 if (mi == pool.mapTx.end())
                     continue;
-                nConflictingFees += mi->GetFeesWithDescendants();
-                nConflictingSize += mi->GetSizeWithDescendants();
+                pool.CalculateDescendants(mi, allConflicting);
```

Comment:
Indeed it doesn't... The more I think about it, the more I think dirty descendants should have been handled by removing such txs from the mempool temporarily, and then adding them back. It only happens during reorgs after all, and we might want to use such a mechanism to help get txs remined during reorgs.